core/services/keeper/orm.go

```diff
@@ -0,0 +1,109 @@
+package keeper
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+const gasBuffer = int32(200_000)
+
+func NewORM(db *gorm.DB) KeeperORM {
+	return KeeperORM{
+		DB: db,
+	}
+}
+
+type KeeperORM struct {
```

Comment:
nit: idiomatic to just call this ORM, the package namespace provides the keeper part. See "Avoid Stutter" here https://blog.golang.org/package-names